### PR TITLE
support inline subworkflow definition

### PR DESCRIPTION
**Pull request by jun-he**
This PR adds the inline subworkflow definition support to subworkflow task.

### CODE REVIEW

**Comment by kishorebanala** [comment]
@jun-he I've been testing locally and were able to successfully verify workflow executions. The UI fails to load parent workflow though. Looking into UI rendering at the moment.

**Comment by kishorebanala** [comment]
@jun-he This looks good to me. Please address the comment I had, and rebase against latest dev branch to merge this.

Also, The UI fails to render Parent workflow when using dynamic Subworkflow definition, as the current logic tries to fetch Subworkflow definition from the API for a subWorkflowTask. Tweaking the UI logic to not fetch when using dynamic Subworkflow definition should do.

**Comment by jun-he** [comment]
@kishorebanala Thanks for the comments. I have addressed them and also rebased the branch.

File: common/src/test/java/com/netflix/conductor/common/workflow/SubWorkflowParamsTest.java [hunk 1]
```diff
@@ -38,4 +39,11 @@ public void testWorkflowSetTaskToDomain() {
...
         assertEquals(taskToDomain, subWorkflowParams.getTaskToDomain());
     }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testSetWorkflowDefinition() {
+        SubWorkflowParams subWorkflowParams = new SubWorkflowParams();//name is null
+        subWorkflowParams.setName("dummy-name");
+        subWorkflowParams.setWorkflowDefinition(new Object());
+    }
 }
```
[the remark this answers is not in the document]
Comment (jun-he) [reply common/src/test/java/com/netflix/conductor/common/workflow/SubWorkflowParamsTest.java:126]: Done.

**Pull request merged**